Convert WorkingWithAlignedNucleotides from Rnw to Rmd

Chisomgold opened the pull request:
Hi @jwokaty 
I've made the sweave to rmd conversion for the WorkingWithAlignedNucleaotides Vignette

Here's the [PDF](https://bioconductor.org/packages/3.16/bioc/vignettes/GenomicAlignments/inst/doc/WorkingWithAlignedNucleotides.pdf) and [HTML](https://github.com/Bioconductor/GenomicAlignments/files/9934824/WorkingWithAlignedNucleotides.zip) for comparison

jwokaty commented on vignettes/WorkingWithAlignedNucleotides.Rmd:
```diff
@@ -0,0 +1,223 @@
+- name: "Hervé Pagès"
+- name: "Halimat C. Atanda"
+  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
+date: "`r format(Sys.Date(), '%B %d %Y')`"
```
Let's follow the language in the PDF by doing

```
date: "Last modified: January 2014; Compiled: `r format(Sys.Date(), '%B %d %Y')`"
```

jwokaty commented on DESCRIPTION:
```diff
@@ -13,8 +13,18 @@ BugReports: https://github.com/Bioconductor/GenomicAlignments/issues
 Encoding: UTF-8
-Author: Hervé Pagès, Valerie Obenchain, Martin Morgan
-Maintainer: Bioconductor Package Maintainer <maintainer@bioconductor.org>
+Authors@R: c(
+    person("Hervé", "Pages", role = "aut"),
+    person("Valerie", "Obenchain", role="aut"),
+    person("Martin", "Morgan", role="aut"),
+    person("Halimat C.", "Atanda", role = "ctb",
+        comment = "Translated 'WorkingWithAlignedNucleotides' vignette from Sweave to RMarkdown / HTML."
+    ),
+    person(
+        "Bioconductor Package Maintainer",
+        role = "cre"
+        email = "maintainer@bioconductor.org"
+    ))
```
The `DESCRIPTION` should follow what's currently in the master branch:

```
Authors@R: c(
	person("Hervé", "Pagès", role=c("aut", "cre"),
		email="hpages.on.github@gmail.com"),
	person("Valerie", "Obenchain", role="aut"),
	person("Martin", "Morgan", role="aut"))
```
Note: it should also use the same indent and spacing around `=`. Herve is the maintainer denoted by `cre` in `role`. You should add your person entry to this following the same format. This will also resolve the merge conflict.

Chisomgold replied:
Hi @jwokaty Should I remove the entry of Bioconductor package maintainer?

jwokaty replied:
Yes, since Herve is the maintainer.

Chisomgold replied:
All done

jwokaty commented on vignettes/WorkingWithAlignedNucleotides.Rmd:
```diff
@@ -0,0 +1,223 @@
+
+After the reads generated by a high-throughput sequencing experiment have been aligned to a reference genome, the questions that are being asked about these alignments typically fall in two broad categories: **positional only** and **nucleotide-related**.
+
+**Positional only** questions are about the position of the alignments with respect to the reference genome. Note that the position of an alignment is actually better described in terms of genomic ranges (1 range for an alignment with no gaps, 2 or more ranges for an alignment with gaps). Knowing the ranges of the alignments is sufficient to perform common tasks like *read counting* or for *compu ...
```
Let's use

```
`?'coverage-methods'`
```

since you need back ticks or quotes in order to get the help page. There should also be no space.

jwokaty reviewed (changes requested):
This looks pretty good. Additionally, I would strongly recommend fixing the long lines in the .Rmd. In RStudio, you can highlight a paragraph then use `Code > Reflow Comment` to break up long lines so that there's a return at line character limit.

jwokaty commented:
@Chisomgold There was still some issue with the DESCRIPTION. I tried to fix it but maybe it just needs to be resolved. I also fixed the long lines from wrapping and renamed code blocks. Thanks for your work!

@hpages When you have time, can you review and let us know if any further changes are required?

Chisomgold commented:
Thanks @jwokaty
I looked through the changes you made.

hpages commented:
Quite a few problems:

- Not sure why but `R CMD build GenomicAlignments` does not try to build the HTML vignette.

- Also `knit()` fails for me:
    ```
    > knit("WorkingWithAlignedNucleotides.Rmd")
    
    processing file: WorkingWithAlignedNucleotides.Rmd
      |.                                                                     |   2%
       inline R code fragments
    
    Quitting from lines 2-72 (WorkingWithAlignedNucleotides.Rmd) 
    Error in Biocpkg("GenomicAlignments") : could not find function "Biocpkg"
    ```
    Did anybody try to knit this?

- Title, authors, dates, and table of content, are missing in the rendered HTML (obtained with `library(knitr); library(BiocStyle); knit("WorkingWithAlignedNucleotides.Rmd"); markdown::mark_html("WorkingWithAlignedNucleotides.md", "WorkingWithAlignedNucleotides.html")`).

- Other Bioconductor packages typically have the following chunk at the beginning of the body their Rmd vignettes:

        ```{r style, echo=FALSE, results='asis'}
        BiocStyle::markdown()
        ```
    Was this purposely omitted here?

- Sections/subsections should be numbered like in original vignette.

- Why split original code chunk:
    ```
    <<bamfiles>>=
    library(RNAseqData.HNRNPC.bam.chr14)
    bamfiles <- RNAseqData.HNRNPC.bam.chr14_BAMFILES
    names(bamfiles)  # the names of the runs
    @
    ```
    in 2 separate chunks?

        ```{r RNAseqData, message=FALSE} 
        library(RNAseqData.HNRNPC.bam.chr14)
        bamfiles <- RNAseqData.HNRNPC.bam.chr14_BAMFILES
        ```
        ``` {r}
        names(bamfiles) # the names of the runs
        ```
    (there are other chunks that were split in a similar way)

- Why modify chunk labels?

- "Bioconductor" is not a package name so you can't translate `other \Bioconductor{} packages` into ``other `r Biocpkg("Biocnductor")` packages`` (also note the typo in "Biocnductor").

- URL `http://bioconductor.org/packages/release/BiocViews.html#___AssayDomains` was broken in original vignette. It should be `https://bioconductor.org/packages/release/BiocViews.html#___AssayDomain` (singular). Also note that this vignette is very old and we've switched to HTTPS for our website since the original vignette was written. All URLs to our website should now use HTTPS.

Thx

Chisomgold commented:
Hello @hpages 

- The file knits on my end and I've attached the most recent [HTML file](https://github.com/Bioconductor/GenomicAlignments/files/10370395/WorkingWithAlignedNucleotides.zip) - changes not pushed yet

- Title, author name, and dates are all present in the HTML file. 
- All sections/subsections are numbered as in the original vignette
- I split the code chunk to avoid printing out the result of  `library(package)` so that the HTML matches the PDF. Without splitting and adding the `message=FALSE` to the first chunk, the new vignette would be much different from the previous. I hope that explains the split.
- For the beginning chunk, I'm not sure I can answer that except that I followed the sweave2rmd contributor guidelines.
- I've fixed the https and spelling error. Thanks for pointing those out.

@jwokaty can you help respond to the ones I may have answered wrongly? Thanks.

hpages commented:
Thanks @Chisomgold, it's easier for everybody else if you push your changes so we can see how you handled the other issues for which you didn't comment, like `R CMD build GenomicAlignments` not building the vignettes, modification of the chunk labels, etc...

Thanks again

Chisomgold commented:
All changes pushed

hpages commented:
Thanks @Chisomgold.

Your `origin/WorkingWithAlignedNucleotides-Rmd` branch is at version 1.33.1 but the current `master` branch is at version 1.35.0. This suggests that you forked and created the `origin/WorkingWithAlignedNucleotides-Rmd` branch off the `master` branch of your fork a long time ago (i.e. before the last BioC release in October). I think this is why GitHub displays the "This branch has conflicts that must be resolved" warning above.

It would be great if you could resolve those conflicts on your side. I'm no git/GitHub expert but I think this involves things like: resyncing your fork, merging master into your branch (this will require resolving the conflicts), and push the branch. Or something like that. Hopefully @jwokaty can provide some guidance.

Other than that, some of the issues I reported earlier still persist:

- `R CMD build GenomicAlignments` still does not build the HTML vignette:
    ```
    hpages@spectre:~/sandbox/GenomicAlignments$ git pull --all
    Fetching origin
    Already up to date.
    
    hpages@spectre:~/sandbox/GenomicAlignments$ git status
    On branch WorkingWithAlignedNucleotides-Rmd
    Your branch is up to date with 'origin/WorkingWithAlignedNucleotides-Rmd'.

    hpages@spectre:~/sandbox/GenomicAlignments$ cd ..

    hpages@spectre:~/sandbox$ bdevbuild GenomicAlignments
    * checking for file ‘GenomicAlignments/DESCRIPTION’ ... OK
    * preparing ‘GenomicAlignments’:
    * checking DESCRIPTION meta-information ... OK
    * cleaning src
    * installing the package to build vignettes
    * creating vignettes ... OK
    * cleaning src
    * checking for LF line-endings in source and make files and shell scripts
    * checking for empty or unneeded directories
    * building ‘GenomicAlignments_1.33.1.tar.gz’
    ```
    Then:
    ```
    tar ztf GenomicAlignments_1.33.1.tar.gz | grep 'doc/WorkingWithAlignedNucleotides'
    ```
    produces no ouput!

- `knit()` still fails for me (see my previous post about this). Of course I can do `library(BiocStyle)` before calling `knit()` to manually work around the problem but I shouldn't need to do this. `knit()` should work out-of-the-box in a fresh R session right after loading the **knitr** package.

- I still don't get the title, authors, dates, and table of content, in the rendered HTML. Also the sections/subsections are not numbered for me. Here is how I generate it:
    ```
    library(knitr)
    library(BiocStyle)
    knit("WorkingWithAlignedNucleotides.Rmd")
    markdown::mark_html("WorkingWithAlignedNucleotides.md", "WorkingWithAlignedNucleotides.html")
    ```
    What am I doing wrong?

- Thanks for explaining the reason why you split the code chunks. I don't think it's necessary to do that to avoid printing out the result of `library(package)`. Just adding `message=FALSE` to the chunk should be enough.

- Also why modify the chunk labels?

Thanks

jwokaty commented:
@hpages Thanks for sharing how you're checking for the html. I've been knitting the file in RStudio, so I didn't notice there was an issue.

I tried your method for generating the html on another package BiocParallel, but it also didn't generate the html with BiocStyle, the contents, etc. This seems to work:

```
rmarkdown::render("vignettes/WorkingWithAlignedNucleotides.Rmd")
```
Regarding the code block renaming, our current method for converting .Rnw to .Rmd doesn't preserve the code block names. I added names rather referring to the original document. I can fix these.

@Chisomgold I noticed that the `DESCRIPTION` file has `VignettedBuilder: knitr` rather than `VignetteBuilder: knitr`. Once that's changed, `R CMD build` produces the html. Since we used RStudio's knit function, I guess it will knit regardless of whether the DESCRIPTION is correct.

Let's try to fix the conflict with this branch by [syncing](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-branch-from-the-command-line) and merging master into WorkingWithAlignedNucleotides-Rmd. During the merge step, you'll need to resolve the conflict in the DESCRIPTION file by choosing the right lines to keep.

hpages commented:
> This seems to work:
>
> ```
> rmarkdown::render("vignettes/WorkingWithAlignedNucleotides.Rmd")
> ```

Ah I see. Good to know. Thx @jwokaty !

Chisomgold commented:
I think the merge conflicts are fixed now.
Should I still merge all code chunks with the `message = FALSE` in R?

jwokaty commented:
@Chisomgold Thanks for taking care of the conflict!

> Should I still merge all code chunks with the `message = FALSE` in R?

Yes, add that to the code chunks where it is needed then push your changes.

Chisomgold commented:
I've done that

hpages commented:
@Chisomgold @jwokaty Where do we stand on this? I still see a red "Changes requested" label above.

hpages commented:
It's actually below.

Chisomgold commented:
To the best of my knowledge, I've made all the requested changes. Could @jwokaty point out the specific changes if something is still missing?

jwokaty commented:
> To the best of my knowledge, I've made all the requested changes. Could @jwokaty point out the specific changes if something is still missing?

I believe it's just the indent for the Authors in the DESCRIPTION file and Robert is also missing, which is also causing the conflict: https://github.com/Bioconductor/GenomicAlignments/blob/devel/DESCRIPTION. 

I can approve the changes after this.

Chisomgold commented:
> > To the best of my knowledge, I've made all the requested changes. Could @jwokaty point out the specific changes if something is still missing?
> 
> I believe it's just the indent for the Authors in the DESCRIPTION file and Robert is also missing, which is also causing the conflict: https://github.com/Bioconductor/GenomicAlignments/blob/devel/DESCRIPTION.
> 
> I can approve the changes after this.

Hello @jwokaty , 
I think I've made the adjustments

jwokaty commented on DESCRIPTION:
```diff
@@ -17,7 +17,11 @@ Authors@R: c(
 	person("Valerie", "Obenchain", role="aut"),
-	person("Martin", "Morgan", role="aut"))
+	person("Martin", "Morgan", role="aut"),
+	person("Robert", "Castelo", role="ctb"
```
I'm sorry, I approved a little too quickly. I noticed we're missing the closing parenthesis and comma for Robert's entry.

Chisomgold commented:
Gosh. So sorry. I've added the paranthesis and comma.

jwokaty commented:
No worries! Thanks for the quick changes!

@hpages Please review.

hpages commented:
Looks good to me. Merging now. Thanks for you work @Chisomgold !
